src/main/java/com/atex/ChangeAspectBeanType.java

```diff
@@ -0,0 +1,450 @@
+  private static int total = 0;
+  private static long timeStarted = 0;
+
+  private static boolean restore = false;
+  private static AtomicInteger restored = new AtomicInteger();
+
+  private static JsonDocument getItem(String id) {
+    JsonDocument response = null;
+    try {
+      response = bucket.get(id);
+    } catch (NoSuchElementException e) {
+      log.warning("No element with message: "
+          + e.getMessage());
+      e.printStackTrace();
+    }
+    return response;
+  }
+
+  private static boolean alreadyConverted(String id) {
+    synchronized (convertedKeys) {
+      if (convertedKeys.contains(id)) return true;
+      convertedKeys.add(id);
+      return false;
+    }
+  }
+
+  private static void execute() throws Exception {
+
+    String filename = "change-aspect-bean-type-" + new Date().getTime() + ".log";
+    FileHandler fileHandler = new FileHandler(filename);
+    SimpleFormatter simple = new SimpleFormatter();
+    fileHandler.setFormatter(simple);
+    log.addHandler(fileHandler);
+    log.setUseParentHandlers(false);
+
+    log.info ("Started @ " + new Date());
+    log.info("Couchbase node: "+cbAddress);
+
+    CouchbaseEnvironment env = DefaultCouchbaseEnvironment.builder()
+            .connectTimeout(TimeUnit.SECONDS.toMillis(60L))
+            .kvTimeout(TimeUnit.SECONDS.toMillis(60L))
+            .viewTimeout(TimeUnit.SECONDS.toMillis(1200L))
+            .maxRequestLifetime(TimeUnit.SECONDS.toMillis(1200L))
+            .autoreleaseAfter(5000)
+            .build();
+
+    Cluster cluster = CouchbaseCluster.create(env, cbAddress);
+
+    try {
+      bucket = cluster.openBucket(cbBucket, cbBucketPwd);
+    } catch (Exception e) {
+      cluster.authenticate("cmuser", cbBucketPwd);
+      bucket = cluster.openBucket(cbBucket);
+    }
+
+    process();
+
+    bucket.close();
+    cluster.disconnect();
+
+    log.info ("Finished @ " + new Date());
+
+    showStatistics();
+
+  }
+
+  private static void process() throws InterruptedException {
+    ViewQuery query;
+    if (devView) {
+      query = ViewQuery.from(design, view).development();
+    } else {
+      query = ViewQuery.from(design, view);
+    }
+    if (startKey != null) {
+      query = query.startKey(startKey);
+    }
+    if (limit > 0) {
+      query.limit(limit);
+    }
+    if (skip > 0) {
+      query.skip(skip);
+    }
+    //query.stale(Stale.FALSE);
+
+    ViewResult result = bucket.query(query);
+    total = result.totalRows();
+    log.info("Number of Hangers in the view : " + total);
+    log.info("limit : " + limit);
+    log.info("skip : " + skip);
+    if (limit > 0 && total > limit) {
+      total = limit;
+    }
+    log.info("Number of Hangers to process : " + total);
+    log.info("Number of Threads: " + numThreads);
+    timeStarted = System.currentTimeMillis();
+
+    ExecutorService executor = Executors.newFixedThreadPool(numThreads);
+    BoundedExecutor bex = new BoundedExecutor(executor, numThreads);
```

Comment:
Why? you already have a fixed thread pool executor. The additional semaphore primitive will just slow down.

Reply:
This is the same pattern as the DeleteOrphans class, I will remove this as you are right; it is not necessary